helper/winsvc/windows_service_windows_test.go

```diff
@@ -0,0 +1,642 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package winsvc
+
+import (
+	"io/fs"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/nomad/ci"
+	"github.com/shoenig/test/must"
+	"golang.org/x/sys/windows/registry"
+	"golang.org/x/sys/windows/svc"
+	"golang.org/x/sys/windows/svc/mgr"
+)
+
+func TestWindowsServiceManager(t *testing.T) {
+	ci.Parallel(t)
+
+	t.Run("IsServiceRegistered", func(t *testing.T) {
+		ci.Parallel(t)
+		t.Run("service does not exist", func(t *testing.T) {
+			ci.Parallel(t)
+			_, manager, closer := makeManagers()
+			defer closer()
+
+			result, err := manager.IsServiceRegistered("fake-service-name")
+			must.NoError(t, err, must.Sprint("check should not error"))
+			must.False(t, result, must.Sprint("service should not exist"))
+		})
+
+		t.Run("service does exist", func(t *testing.T) {
+			ci.Parallel(t)
+			m, manager, closer := makeManagers()
+			defer closer()
+			serviceName, closer := generateStubService(m)
+			defer closer()
+
+			result, err := manager.IsServiceRegistered(serviceName)
+			must.NoError(t, err, must.Sprint("check should not error"))
+			must.True(t, result, must.Sprint("service should exist"))
+		})
+	})
+
+	t.Run("GetService", func(t *testing.T) {
+		ci.Parallel(t)
+		t.Run("service does not exist", func(t *testing.T) {
+			ci.Parallel(t)
+			_, manager, closer := makeManagers()
+			defer closer()
+			_, err := manager.GetService("fake-service-name")
+			must.Error(t, err, must.Sprint("error should be generated when service does not exist"))
```

Comment:
I like to assert a specific error, either with `must.ErrorIs` if I have an error "type" (`var ErrWhatever = errors.New("whatever")`), or with `must.EqError` or `must.ErrorContains` for message string/substring checking.

maybe it doesn't matter in this case, because this func only has the single error code path, but again this is a habit thing, so I just always check for something specific.